File: snapshots/input/syntax/Main/Operators.cs
using System.Diagnostics.CodeAnalysis;

namespace Main;

[SuppressMessage("ReSharper", "all")]
public class Operators
{
    class PlusMinus
    {
        public static int operator +(PlusMinus a)
        {
            return 0;
        }

        public static int operator +(PlusMinus a, PlusMinus b)
        {
            return 0;
        }

        public static int operator -(PlusMinus a)
        {
            return 0;
        }
    }

    class TrueFalse
    {
        protected bool Equals(TrueFalse other)
        {
            throw new NotImplementedException();
        }

        public override bool Equals(object? obj)
        {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            if (obj.GetType() != this.GetType()) return false;
            return Equals((TrueFalse)obj);
        }

        public override int GetHashCode()
        {
            throw new NotImplementedException();
        }

        public static bool operator true(TrueFalse a)
        {
            return true;
        }

        public static bool operator false(TrueFalse a)
        {
            return false;
        }

        public static bool operator !=(TrueFalse a, TrueFalse b)
        {
            return true;
        }

        public static bool operator ==(TrueFalse a, TrueFalse b)
        {
            return true;
        }
    }
}

File: snapshots/input/syntax/VBMain/Operators.vb
Imports System.Diagnostics.CodeAnalysis

Namespace VBMain
    <SuppressMessage("ReSharper", "all")>
    Public Class Operators
        Public Class PlusMinus
            Public Shared Operator +(A As PlusMinus)
                Return 0
            End Operator

            Public Shared Operator +(A As PlusMinus, B As PlusMinus)
                Return 0
            End Operator

            Public Shared Operator -(A As PlusMinus)
                Return 0
            End Operator
        End Class

        Public Class TrueFalse
            Public Shared Operator IsTrue(A As TrueFalse) As Boolean
                Return True
            End Operator

            Public Shared Operator IsFalse(A As TrueFalse) As Boolean
                Return False
            End Operator

            Public Shared Operator =(A As TrueFalse, B As TrueFalse) As Boolean
                Return True
            End Operator

            Public Shared Operator <>(A As TrueFalse, B As TrueFalse) As Boolean
                Return True
            End Operator

            Public Overrides Function Equals(obj As Object) As Boolean
                If ReferenceEquals(Nothing, obj) Then Return False
                If ReferenceEquals(Me, obj) Then Return True
                If obj.GetType() <> Me.GetType() Then Return False
                Return Equals(CType(obj, TrueFalse))
            End Function

            Public Overrides Function GetHashCode() As Integer
                Throw New NotImplementedException()
            End Function

        End Class
    End Class
End Namespace
